src/docfx/lib/HtmlUtility.cs

```diff
@@ -325,6 +325,44 @@ public static string CreateHtmlMetaTags(JObject metadata)
         return result.ToString();
     }
 
+    public static void CollectHtmlUsage(string html, Dictionary<string, Dictionary<string, int>> elementCount)
+    {
+        var reader = new HtmlReader(html);
+
+        while (reader.Read(out var token))
+        {
+            CollectHtmlUsageCore(ref token, elementCount);
+        }
+
+        static void CollectHtmlUsageCore(ref HtmlToken token, Dictionary<string, Dictionary<string, int>> elementCount)
+        {
+            if (token.Type != HtmlTokenType.StartTag)
+            {
+                return;
+            }
+
+            var tokenName = token.Name.ToString();
+            if (!elementCount.ContainsKey(tokenName))
```

Comment:
use `TryGetValue`

Reply:
I thought about using `TryGetValue` here, but if the dictionary does not have contain the key, we still need to add (key, dictionary<>) pair.

Reply:
But you can still save one access to the dictionary if the key exists.

Reply:
Maybe we could have a helper like this: https://github.com/dotnet/runtime/issues/15059#issuecomment-954708195